### PR TITLE
Allow modules to mint on their own authority

### PR DESCRIPTION
**Pull request by preston-evans98**
# Description
IBC requires modules to be able to mint tokens on their own authority, rather than using `context.sender()` as the authorizer of the mint. This change enables that functionality.

### CODE REVIEW

File: module-system/module-implementations/sov-bank/src/token.rs [hunk 1]
```diff
@@ -164,14 +164,14 @@ impl<C: sov_modules_api::Context> Token<C> {
...
     }
 
-    /// Mints a given `amount` of token sent by `sender` to the specified `minter_address`.
+    /// Mints a given `amount` of token sent by `sender` to the specified `mint_to_address`.
     /// Checks that the `authorized_minters` set is not empty for the token and that the `sender`
-    /// is an `authorized_minter`. If so, update the balances of token for the `minter_address` by
+    /// is an `authorized_minter`. If so, update the balances of token for the `mint_to_address` by
     /// adding the minted tokens. Updates the `total_supply` of that token.
-    pub(crate) fn mint(
+    pub(crate) fn mint_from_eoa(
```
Comment (plafer) [inline module-system/module-implementations/sov-bank/src/token.rs]: To follow similar naming to bank, shouldn't this be `mint`, and `sender` argument should be `authorizer`? Given that we're trying to enable "sender" to not always be the EOA that initiated the transaction?

**Review by plafer — COMMENTED** [review]
Semantics LGTM, left a comment/question about naming

File: module-system/module-implementations/sov-bank/src/token.rs (hunk 1 above)
Comment (preston-evans98) [reply module-system/module-implementations/sov-bank/src/token.rs]: Oops, yeah I did an automatic refactor here and didn't review. Fixing!

File: module-system/module-implementations/sov-bank/src/call.rs [hunk 2]
```diff
@@ -137,29 +137,41 @@ impl<C: sov_modules_api::Context> Bank<C> {
...
     /// Calls the [`Token::mint`] function and update the `self.tokens` set to store the new minted address.
     pub fn mint(
         &self,
         coins: &Coins<C>,
-        minter_address: &C::Address,
-        context: &C,
+        mint_to_address: &C::Address,
+        authorizer: &C::Address,
         working_set: &mut WorkingSet<C::Storage>,
     ) -> Result<CallResponse> {
```
Comment (bkolad) [inline module-system/module-implementations/sov-bank/src/call.rs]: Does it have to return `CallResponse`, now it is a public fn so I think `()` should be enough?

**Review by bkolad — APPROVED** [review]

File: module-system/module-implementations/sov-bank/src/call.rs [hunk 3]
```diff
@@ -137,29 +137,41 @@ impl<C: sov_modules_api::Context> Bank<C> {
...
         let context_logger = || {
             format!(
                 "Failed mint coins({}) to {} by minter {}",
-                coins,
-                minter_address,
-                context.sender()
+                coins, mint_to_address, authorizer
```
Comment (citizen-stig) [inline module-system/module-implementations/sov-bank/src/call.rs:167]: Should message be updated to `by authorizer` ?

File: module-system/module-implementations/sov-bank/src/call.rs [hunk 4]
```diff
@@ -137,29 +137,41 @@ impl<C: sov_modules_api::Context> Bank<C> {
         Ok(CallResponse::default())
     }
 
-    /// Mints the `coins` set by the address `minter_address`. If the token address doesn't exist return an error.
+    /// Mints the `coins`to the address `mint_to_address` using `context.sender()` as the authorizer.
+    /// Returns an error if the token address doesn't exist or `context.sender()` is not authorized to mint tokens.
+    /// Calls the [`Token::mint`] function and update the `self.tokens` set to store the new balance.
+    pub fn mint_from_eoa(
```
Comment (citizen-stig) [inline module-system/module-implementations/sov-bank/src/call.rs:144]: super minor nit, but it would be good to say what does "EOA" stands for.

**Review by citizen-stig — APPROVED** [review]

File: module-system/module-implementations/sov-bank/src/call.rs (hunk 3 above)
Comment (preston-evans98) [reply module-system/module-implementations/sov-bank/src/call.rs:167]: Fixed.

File: module-system/module-implementations/sov-bank/src/call.rs (hunk 4 above)
Comment (preston-evans98) [reply module-system/module-implementations/sov-bank/src/call.rs:144]: Done!

File: module-system/module-implementations/sov-bank/src/call.rs (hunk 2 above)
Comment (preston-evans98) [reply module-system/module-implementations/sov-bank/src/call.rs]: I'll change to `()`.

**Pull request merged**